#!/usr/bin/env python
PACKAGE = 'laser_processor'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('min_cluster_size',     int_t,      0, '', 20, 1, 10000)
gen.add('max_cluster_size',     int_t,      0, '', 2000, 1, 20000)
gen.add('cluster_tolerance',    double_t,   0, 'm', 0.15, 0, 10)

gen.add('leaf_size_x',          double_t,   0, 'm', 0.02, 0, 1.0)
gen.add('leaf_size_y',          double_t,   0, 'm', 0.02, 0, 1.0)
gen.add('leaf_size_z',          double_t,   0, 'm', 0.02, 0, 1.0)

gen.add('mean_k',               int_t,      0, '', 10, 1, 10000)
gen.add('std_dev_mult',         double_t,   0, '', 1.0, 0, 10.0)

gen.add('probability',         double_t,   0, '', 0.5, 0, 1)

exit(gen.generate(PACKAGE, 'laser_processor', 'LaserProcessor'))
